kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/batch/GCSBatchTableWriter.java

```diff
@@ -0,0 +1,148 @@
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Batch Table Writer that uploads records to GCS as a blob
+ * and then triggers a load job from that GCS file to BigQuery.
+ */
+public class GCSBatchTableWriter implements Runnable {
+  private static final Gson gson = new Gson();
+  private static final int NOT_FOUND_ERROR_CODE = 404;
+
+  private final BlobInfo blobInfo;
+  private final Storage storage;
+  private final String sourceUri;
+
+  private final BigQuery bigQuery;
+  private final LoadJobConfiguration loadJobConfiguration;
+
+  private final List<Map<String, Object>> records;
+
+  /**
+   * Initializes a batch table writer with a full list of records to write.
+   * @param bucketName The name of the bucket to upload the blob in
+   * @param blobName Full path within the bucket to the blob (without the extension)
+   * @param storage GCS Storage
+   * @param tableId {@link TableId} of the BigQuery table to upload to
+   * @param bigQuery {@link BigQuery} Object used to perform upload
+   * @param records Records to upload to BigQuery through GCS
+   */
+  public GCSBatchTableWriter(String bucketName,
+                             String blobName,
+                             Storage storage,
+                             TableId tableId,
+                             BigQuery bigQuery,
+                             List<Map<String, Object>> records) {
+    BlobId blobId = BlobId.of(bucketName, blobName + ".json");
+    blobInfo = BlobInfo.newBuilder(blobId).setContentType("text/json").build();
+    this.storage = storage;
+    this.sourceUri = String.format("gs://%s/%s.json", bucketName, blobName);
+
+    this.bigQuery = bigQuery;
+
+    // Check if the table specified exists
+    // This error shouldn't be thrown. All tables should be created by the connector at startup
+    if (bigQuery.getTable(tableId) == null || !bigQuery.getTable(tableId).exists()) {
+      throw new ConnectException("Table with TableId %s does not exist.");
+    }
+
+    this.loadJobConfiguration =
+        LoadJobConfiguration.builder(tableId, sourceUri)
+                            .setFormatOptions(FormatOptions.json())
+                            .setCreateDisposition(JobInfo.CreateDisposition.CREATE_NEVER)
+                            .build();
+
+    this.records = records;
+  }
+
+  @Override
+  public void run() {
+    //todo implement
+  }
+
+  /**
+   * Triggers a load job to transfer JSON records from a GCS blob to a BigQuery table.
+   */
+  private void triggerBigQueryLoadJob() {
+    Job loadJob = bigQuery.create(JobInfo.of(loadJobConfiguration));
+    try {
+      loadJob.waitFor();
+    } catch (InterruptedException | TimeoutException exception) {
+      String exceptionMessage = String.format("%s.Source URI = \"%s\" Table = \"%s\"",
+          "Transfer from GCS blob to BigQuery unsuccessful.",
```

Comment:
I'm a little confused here. Why not just have this in the initial string being formatted?